scripts/data/src/org/akvo/gae/remoteapi/FixOldTranslations.java

```diff
@@ -0,0 +1,129 @@
+        Long parentId = (Long) t.getProperty("parentId");
+        return fixTranslation(t, ds, parentId);
+    }
+
+    private Entity fixQuestionTranslation(Entity t, final DatastoreService ds) {
+        System.out.println("Fixing Question translation");
+        Long parentId = (Long) t.getProperty("parentId");
+        if (parentId != null) {
+            final Entity question = fetchQuestion(ds, parentId);
+            if (question != null) {
+                return fixTranslation(t, ds, (Long) question.getProperty("questionGroupId"));
+            }
+        }
+        return null;
+    }
+
+    private Entity fixQuestionOptionTranslation(Entity t, final DatastoreService ds) {
+        System.out.println("Fixing QuestionOption translation");
+        Long parentId = (Long) t.getProperty("parentId");
+        if (parentId != null) {
+            final Query.Filter fs = new Query.FilterPredicate(Entity.KEY_RESERVED_PROPERTY,
+                    Query.FilterOperator.EQUAL, KeyFactory.createKey("QuestionOption", parentId));
+            final Query q = new Query("QuestionOption").setFilter(fs);
+            Entity option = ds.prepare(q).asSingleEntity();
+            if (option != null) {
+                final Entity question = fetchQuestion(ds, (Long) option.getProperty("questionId"));
+                if (question != null) {
+                    return fixTranslation(t, ds, (Long) question.getProperty("questionGroupId"));
+                }
+            }
+        }
+        return null;
+    }
+
+    private Entity fixTranslation(Entity t, DatastoreService ds, Long groupId) {
+        final Entity qGroup = fetchGroup(ds, groupId);
+        if (qGroup != null) {
+            t.setProperty("surveyId", qGroup.getProperty("surveyId"));
+            t.setProperty("questionGroupId", groupId);
+            return t;
+        }
+        return null;
+    }
+
+    private List<Entity> fetchBrokenTranslations(DatastoreService ds) {
+        List<Entity> translations = new ArrayList<>();
+        final Query q = new Query("Translation");
+        final PreparedQuery pq = ds.prepare(q);
+        final Query.Filter filterNullSurveyId = new Query.FilterPredicate("surveyId", Query.FilterOperator.EQUAL, null);
+        final Query.Filter filterNullQuestionGroup = new Query.FilterPredicate("questionGroupId", Query.FilterOperator.EQUAL, null);
+        q.setFilter(Query.CompositeFilterOperator.or(filterNullSurveyId, filterNullQuestionGroup));
+        List<Entity> entities = pq.asList(FetchOptions.Builder.withChunkSize(500));
+        translations.addAll(entities);
+        System.out.println("Found " + translations.size() + " broken translations");
+        return translations;
+    }
+
+    private Entity fetchGroup(DatastoreService ds, Long groupId) {
+        if (groupId == null) {
+            return null;
+        }
+        final Query.Filter fs = new Query.FilterPredicate(Entity.KEY_RESERVED_PROPERTY,
```

Comment:
If you know the `id` of the entity you want, you can just do `.get()`, you don't need to *query* for it:
You need to convert the `Long` value into ` Key`:
https://cloud.google.com/appengine/docs/standard/java/datastore/entities#Retrieving_an_entity

You can get a `Key` passing the `Long` value and the `Kind` name:
https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/KeyFactory.html#createkey_2